library/lua/custom-raw-tokens.lua

```diff
@@ -0,0 +1,243 @@
+--[[
+custom-raw-s
+Allows for reading custom tokens added to raws by mods
+by Tachytaenius (wolfboyft)
+
+Yes, custom raw tokens do quietly print errors into the error log but the error log gets filled with garbage anyway
+]]
+
+local _ENV = mkmodule("custom-raw-tokens")
+
+local eventful = require("plugins.eventful")
+local utils = require("utils")
+
+local customRawTokensCache = {}
+eventful.onUnload["custom-raw-tokens"] = function()
+    customRawTokensCache = {}
+end
+
+-- Not state, defined later:
+local ensureTable, doTable, getSubtype, rawStringsFieldNames
+local getTokenCore, getRaceCasteTokenCore
```

Comment:
this is not a common approach, and I don't think I can recommend it. It is clearer (and more common) to define your locals (vars and functions) above the code that uses them.